[FLINK-20915][docker] Move docker entrypoint to distribution

zentol opened the pull request:
Counterpart to https://github.com/apache/flink/pull/14630.

CI will naturally fail on this branch since the Flink snapshots don't contain the necessary changes yet. Please refer to the CI run in the linked PR.

rmetzger commented on docker-entrypoint.sh:
```diff
@@ -41,60 +31,6 @@ drop_privs_cmd() {
-
 disable_jemalloc_env() {
   # use nameref '_args' to update the passed 'args' within function
   local -n _args=$1
```
COMMAND_DISABLE_JEMALLOC is not defined in this file (anymore)

rmetzger commented on docker-entrypoint.sh:
```diff
@@ -107,80 +43,10 @@ disable_jemalloc_env() {
-. $FLINK_HOME/bin/config.sh
-export FLINK_CLASSPATH="`constructFlinkClassPath`:$INTERNAL_HADOOP_CLASSPATHS"
-
 # Running command in pass-through mode
```
```suggestion
# Running command in pass-through mode: The actual docker-entrypoint.sh logic is in the Flink distribution.
```

rmetzger reviewed (changes requested):
I'll comment in the other PR.

rmetzger commented on docker-entrypoint.sh:
```diff
@@ -41,60 +33,6 @@ drop_privs_cmd() {
-    fi
-    envsubst < "${CONF_FILE}" > "${CONF_FILE}.tmp" && mv "${CONF_FILE}.tmp" "${CONF_FILE}"
-}
-
```
@wangyang0918 mentioned that he'd prefer controlling jemalloc via an environment variable

zentol replied:
Out-of-scope of this PR as far as I'm concerned. That's an API change that I would move to a follow-up.

wangyang0918 replied:
I believe setting the env(`export DISABLE_JEMALLOC=true`) is more convenient to use and could be easily integrated in all modes(docker, standalone, native-k8s). I agree that this could be done as a follow-up.

rmetzger replied:
Okay, let's do it later then

rmetzger reviewed (approved):
Thanks for addressing this.

zentol commented:
I have updated the PR. The previous version broke the jemalloc switch since we were checking the wrong argument index. I did not want to deal with mutating arrays, and instead refactored the switch to a environment variable and added tests for the behavior.
